bee-tangle/src/solid_entry_point.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! A SolidEntryPoint is a [`MessageId`](crate::MessageId) of a message even if we do not have them
+//! or their past in the database. They often come from a snapshot file and allow a node to solidify
+//! without needing the full tangle history.
+
+use bee_common::packable::{Packable, Read, Write};
+use bee_message::MessageId;
+
+use ref_cast::RefCast;
+
+use core::{convert::AsRef, ops::Deref};
+
+/// A SolidEntryPoint is a [`MessageId`] of a message even if we do not have them
+/// or their past in the database. They often come from a snapshot file and allow a node to solidify
+/// without needing the full tangle history.
+///
+/// This is a type wrapper around a [`MessageId`](crate::MessageId) to differentiate it from a
+/// non-solidified message.
+///
+/// Spec: #iota-protocol-rfc
+/// <https://github.com/iotaledger/protocol-rfcs/blob/master/text/0005-white-flag/0005-white-flag.md>
+#[derive(RefCast)]
+#[repr(transparent)]
```

Comment:
I know there was a discussion about this, should it stay then ?

Reply:
It's needed for `RefCast`

Reply:
Thanks